### DISCUSSION TITLE
appDir on Vercel: TypeError: Cannot read properties of undefined (reading 'map')

### DISCUSSION THREAD

**Discussion by dluan** [vercel/next.js#46063] [Help]
An error is occurring, and the feedback is not very clear on how I should resolve it.
When I run 'yarn build' sometimes it works, other times it doesn't with the same code.

```
// package.json
{
...
"next": "13.1.1"
react": "18.2.0"
...
}
```

```
// app/page.tsx
...
async function Carousel() {
   const { Banner } = await request<BannerResponse>(GET_BANNER)

   return <Hero banners={Banner} data-superjson />
}

export default function Home() {
   return (
      <>
         <div>
            <Suspense fallback={<SkeletonHero />}>
               {/* @ts-expect-error Server Component */}
               <Carousel />
            </Suspense>
...
```

```
// components/home/Hero.tsx
'use client'
...
export default function Hero({ banners }: HeroProps) {
   return (
      <div className="border-y-2 border-indigo-600 max-h-[70vh] overflow-hidden">
         <CarouselProvider
            naturalSlideWidth={1920}
            naturalSlideHeight={652}
            totalSlides={banners ? banners.length : 2}
            infinite={true}
            dragEnabled={false}
         >
            <Slider>
            {banners.map((banner, index) => (
...
```

The following error appears in the terminal: "TypeError: Cannot read properties of undefined (reading 'map')"
![image](https://user-images.githubusercontent.com/1024699/219667164-2ec65450-55a1-47f0-8f17-b892aedc5815.png)

The same happens when I try to publish on Vercel, the same code may or may not work.

Did I do something wrong?

**Comment by dluan** [comment]
I got a solution, but I think it is necessary to review the cause of the problem further.

I tried the solution from another thread #43942 

When deleting the `.next` folder, the build works normally. And when trying to publish to Vercel, you need to redeploy or set the `VERCEL_FORCE_NO_BUILD_CACHE` environment variable to 1, [as described in the documentation](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#managing-build-cache). That way the code I mentioned here in the discussion works.

**Comment by 96312** [comment]
I get the same shit when deploying to vercel and it works absolutely okay in local development server!

**Reply by dluan** [reply]
> I got a solution, but I think it is necessary to review the cause of the problem further.
> 
> I tried the solution from another thread #43942
> 
> When deleting the `.next` folder, the build works normally. And when trying to publish to Vercel, you need to redeploy or set the `VERCEL_FORCE_NO_BUILD_CACHE` environment variable to 1, [as described in the documentation](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#managing-build-cache). That way the code I mentioned here in the discussion works.

Did you try cleaning the build cache on Vercel?

**Reply by 96312** [reply]
Yes, I did. What ended up working for me was your suggestion and this comment - https://github.com/vercel/next.js/discussions/43942#discussioncomment-4844689

Deploying to vercel isn't as straightforward. But it works now! Thanks.
